.github/workflows/staged.yaml

```diff
@@ -0,0 +1,124 @@
+name: Staged commit
+# This is the full suite of tests which is run on
+# a staged commit, just prior to merging into the
+# main tree.
+# This is meant to be the most comprehensive suite of
+# tests; they are expensive and slow to run and
+# therefore we want to limit our reliance on them.
+#
+# The number of these tests needs to be kept
+# in sync with Anubis configuration
```

Comment:
The suite of tests in this file is not "full" or "comprehensive" AFAICT -- it excludes functionality tests, source maintenance tests, CodeQL tests, and possibly some of Jenkins tests. 

The concern about Anubis configuration update applies to the tests defined in the other YAML file as well AFAICT because those tests are also executed for staged commits.

Similar to the other change request, I recommend removing this misleading description. We can make progress by documenting our changing test placement logic in PR description, at least for now. Once the dust settles, all files are (re)named appropriately, all tests are placed appropriately, we can decide to document our finalized logic in sources.

```suggestion
```

FWIW, right now, I suspect that this file might become something like "extra.yaml" or even "slow.yaml" while the old one might become something like "minimal.yaml" or "quick.yaml" (or even retain its original default.yaml name!).

Reply:
> The suite of tests in this file is not "full" or "comprehensive" AFAICT -- it excludes functionality tests, source maintenance tests, CodeQL tests, and possibly some of Jenkins tests.

Lies, big lies, and carefully chosen metrics on my side.
I'm better clarifying intent that the aim is to check for compiler, OS, and platform
portability issues.

> The concern about Anubis configuration update applies to the tests defined in the other YAML file as well AFAICT because those tests are also executed for staged commits.

Yes. Clarified as well, copying that snippet to all comments.

> Similar to the [other change request](https://github.com/squid-cache/squid/pull/1917#discussion_r1813371727), I recommend removing this misleading description. We can make progress by documenting our changing test placement logic in PR description, at least for now. Once the dust settles, all files are (re)named appropriately, all tests are placed appropriately, we can decide to document our finalized logic in sources.
> 
> FWIW, right now, I suspect that this file might become something like "extra.yaml", "more.yaml", or even "slow.yaml" while the old one might become something like "first.yaml" or "quick.yaml" (or even retain its original default.yaml name!).

Possibly. Whatever it will be, I don't think it's worth delaying the PR on.
As soon as this goes in, I can start merging FreeBSD support which is ready and waiting, and then look into ccache which is mostly ready and only needs
to be fixed in the context of the squid-cache account, and then more platforms which
I have some idea on how to do but requires some tinkering.
When that is done, we can - for the most part if not in full - retire jenkins.